src/components/ChatApp/MessageSection/MessageSection.react.js

```diff
@@ -431,35 +448,34 @@ class MessageSection extends Component {
       speechOutput,
       speechOutputAlways,
       theme,
-      customThemeValue,
       messages,
       messagesByID,
       loadingHistory,
       backgroundImage,
       messageBackgroundImage,
+      customThemeValue,
       loadingReply,
     } = this.props;
-    const { search, searchState, showScrollTop, showScrollBottom } = this.state;
-    const {
-      scrollBottomStyle,
-      scrollTopStyle,
-      messageBackgroundStyles,
-    } = styles;
+    let header, pane, body, composer, button;
+
+    if (theme === 'custom') {
+      header = customThemeValue.header;
+      pane = customThemeValue.pane;
+      body = customThemeValue.body;
+      composer = customThemeValue.composer;
+      button = customThemeValue.button;
+    } else {
+      header = theme === 'dark' ? dark.header : light.header;
```

Comment:
The logic is too complex for it and involves too many conditions across files. Instead set the const object of dark and light theme in `utils/colors` it to customTheme value at the reducer itself.

Reply:
On it